docs/appendix-A.rst

```diff
@@ -0,0 +1,1783 @@
+
+.. literalinclude:: ../regression/reference/vectors/wrapfvectors.f
+   :language: fortran
+   :start-after: start vector_iota_out_alloc
+   :end-before: end vector_iota_out_alloc
+   :dedent: 4
+
+``inta`` is ``intent(out)``, so it will be deallocated upon entry to ``vector_iota_out_alloc``.
+
+Fortran usage:
+
+.. code-block:: fortran
+
+    integer(C_INT), allocatable :: inta(:)
+    call vector_iota_out_alloc(inta)
+    call assert_true(allocated(inta))
+    call assert_equals(5 , size(inta))
+    call assert_true( all(inta == [1,2,3,4,5]), &
+         "vector_iota_out_alloc value")
+
+
+.. ############################################################
+
+.. _example_vector_iota_inout_alloc:
+
+vector_iota_inout_alloc
+^^^^^^^^^^^^^^^^^^^^^^^
+
+.. literalinclude:: ../regression/run/vectors/vectors.cpp
+   :language: c
+   :start-after: start vector_iota_inout_alloc
+   :end-before: end vector_iota_inout_alloc
+
+YAML:
+
+.. code-block:: yaml
+
+    - decl: void vector_iota_out_alloc(std::vector<int> &arg+intent(inout)+deref(allocatable))
+
+The C wrapper:
+
+.. literalinclude:: ../regression/reference/vectors/wrapvectors.cpp
+   :language: c
+   :start-after: start VEC_vector_iota_inout_alloc_bufferify
+   :end-before: end VEC_vector_iota_inout_alloc_bufferify
+
+Calls C via the interface:
+
+.. literalinclude:: ../regression/reference/vectors/wrapfvectors.f
+   :language: fortran
+   :start-after: start c_vector_iota_inout_alloc_bufferify
+   :end-before: end c_vector_iota_inout_alloc_bufferify
+   :dedent: 8
+
+The Fortran wrapper:
+
+.. literalinclude:: ../regression/reference/vectors/wrapfvectors.f
+   :language: fortran
+   :start-after: start vector_iota_inout_alloc
+   :end-before: end vector_iota_inout_alloc
+   :dedent: 4
+
+
+``inta`` is ``intent(inout)``, so it will NOT be deallocated upon
+entry to ``vector_iota_inout_alloc``.
+Fortran usage:
+
+.. code-block:: fortran
+
+    call vector_iota_inout_alloc(inta)
+    call assert_true(allocated(inta))
+    call assert_equals(10 , size(inta))
+    call assert_true( all(inta == [1,2,3,4,5,11,12,13,14,15]), &
+         "vector_iota_inout_alloc value")
+    deallocate(inta)
+
+Void Pointers
+-------------
+
+.. ############################################################
+
+.. _example_passAssumedType:
+
+passAssumedType
+^^^^^^^^^^^^^^^
+
+C library function:
+
+.. literalinclude:: ../regression/run/clibrary/clibrary.c
+   :language: c
+   :start-after: start passAssumedType
+   :end-before: end passAssumedType
+
+YAML:
+
+.. code-block:: yaml
+
+    - decl: int passAssumedType(void *arg+assumedtype)
+
+Calls C via the interface:
+
+.. literalinclude:: ../regression/reference/clibrary/wrapfclibrary.f
+   :language: fortran
+   :start-after: start pass_assumed_type
+   :end-before: end pass_assumed_type
+   :dedent: 8
+
+Fortran usage:
+
+.. code-block:: fortran
+
+    use iso_c_binding, only : C_INT
+    integer(C_INT) rv_int
+    rv_int = pass_assumed_type(23_C_INT)
```

Comment:
What is `23_C_int` ?

Reply:
I added a comment for C++ users.  23_kind is how specific constants are created. In this case, it creates a variable of type integer(C_INT) instead of default integer.  They're usually the same kind. This just makes sure it is a C int.